libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m

```diff
@@ -402,36 +402,34 @@ - (NSString *)sfHybridViewUserAgentString
     return userAgentString;
 }
 
-- (NSURL *)frontDoorUrlWithReturnUrl:(NSString *)returnUrl returnUrlIsEncoded:(BOOL)isEncoded createAbsUrl:(BOOL)createAbsUrl
+- (NSURL *)frontDoorUrlWithReturnUrl:(NSString *)returnUrlString returnUrlIsEncoded:(BOOL)isEncoded createAbsUrl:(BOOL)createAbsUrl
 {
     SFOAuthCredentials *creds = [SFUserAccountManager sharedInstance].currentUser.credentials;
-    NSString *instUrl = creds.apiUrl.absoluteString;
-    NSString *fullReturnUrl = returnUrl;
-
+    NSURL *instUrl = creds.apiUrl;
+    NSString *fullReturnUrlString = returnUrlString;
+    
     /*
      * We need to use the absolute URL in some cases and relative URL in some
      * other cases, because of differences between instance URL and community URL.
      */
-    if (createAbsUrl && ![returnUrl hasPrefix:@"http"]) {
-        fullReturnUrl = [NSString stringWithFormat:@"%@/%@", instUrl, returnUrl];
+    if (createAbsUrl && ![returnUrlString hasPrefix:@"http"]) {
+        fullReturnUrlString = [instUrl URLByAppendingPathComponent:returnUrlString].absoluteString;
     }
-    if([returnUrl containsString:@"frontdoor.jsp"]) {
-        NSRange r1 = [returnUrl rangeOfString: isEncoded ? @"retURL%3D" : @"retURL="];
-        NSRange r2 = [returnUrl rangeOfString: isEncoded ? @"%26display" : @"&display"];
+    if ([returnUrlString containsString:@"frontdoor.jsp"]) {
```

Comment:
In looking through some revamp of this method, is this block of functionality something we can revisit, or remove?  I don't really understand how the [original issue](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/issues/1515) (fix [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1519)) happens—and it seems like an edge case that should be handled in some other way, or is surfacing some other bug.  How do you end up with a return URL that itself is frontdoor.jsp?  That just shouldn't happen.

@bhariharan do you remember anything about this issue?

Reply:
Unfortunately, I don't remember the exact use case. However, if I were to guess, I think it would have something to do with custom logout flows that are configurable in the community context. There are some use cases where this could potentially happen. There's probably no clean way to do this unless we get a first class native API that we can hit for session validity and other related functionality.